packages/http-server/src/server.ts

```diff
@@ -75,14 +91,22 @@ const replyHandler = <LoaderInput>(
           reply.headers(output.headers);
         }
 
-        reply.serializer((payload: unknown) => payload);
-        reply.send(output.body);
+        reply.serializer((payload: unknown) => payload).send(output.body);
       } else {
         reply.code(500).send('Unable to find any decent response for the current request.');
       }
     } catch (e) {
-      reply.code(500).send(e);
+      const status = 'status' in e ? e.status : 500;
+      reply
+        .type('application/problem+json')
+        .serializer(problemJsonSerializer)
+        .code(status)
+        .send({
+          type: e.name || 'https://stoplight.io/prism/errors#UNKNOWN',
```

Comment:
I like this :) can we make all error names follow this syntax?

Reply:
Yes! The idea would also be to create a real webpage with bookmark so that we can also provide suggestions on how to fix it.

Reply:
Fixed in 4a481c8